### PR TITLE
executor_test: improve show full processlist tests

### PR DESCRIPTION
**Pull request by Darren**
test cases added in PR #4739 hardcoded query id, it brings some trouble that each time new test cases added like in #5158  and #5210 we have to modify the hardcoded query id.

This PR fixes this.

@shenli @mccxj

### CODE REVIEW

**Comment by shenli** [comment]
@Darren Thanks!

File: util/testkit/testkit.go [hunk 1]
```diff
@@ -48,6 +48,21 @@ func (res *Result) Check(expected [][]interface{}) {
 	res.c.Assert(got, check.Equals, need, res.comment)
 }
 
+// CheckAt asserts the result of selected columns equals the expected results.
+func (res *Result) CheckAt(cols []int, expected [][]interface{}) {
+	var rows [][]string
```
Comment (shenli) [inline util/testkit/testkit.go]: rows := make([][]string, 0, len(expected))

File: util/testkit/testkit.go [hunk 2]
```diff
@@ -48,6 +48,21 @@ func (res *Result) Check(expected [][]interface{}) {
...
+	for i := range res.rows {
+		var row []string
```
Comment (shenli) [inline util/testkit/testkit.go]: row := make([]string, 0, len(cols))

File: util/testkit/testkit.go (hunk 1 above)
Comment (Darren) [reply util/testkit/testkit.go]: done

**Comment by shenli** [comment]
LGTM

File: util/testkit/testkit.go [hunk 3]
```diff
@@ -48,6 +48,27 @@ func (res *Result) Check(expected [][]interface{}) {
 	res.c.Assert(got, check.Equals, need, res.comment)
 }
 
+// CheckAt asserts the result of selected columns equals the expected results.
+func (res *Result) CheckAt(cols []int, expected [][]interface{}) {
+	for _, e := range expected {
+		if len(cols) != len(e) {
+			panic("lens of cols and expected cols should be equal")
```
Comment (XuHuaiyu) [inline util/testkit/testkit.go]: why use panic here but not res.c.Assert ?

Comment (Darren) [reply util/testkit/testkit.go]: fixed

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**